Test failures building on Windows

This tool looks exactly like what I need to process a whole bunch of artifacts, but first I need to compile it before trying it out.

Unfortunately, using Maven 3.9.9 with Java 21.0.4 on Windows 11 resulted in the following:

```
[INFO]
[INFO] Results:
[INFO]
[ERROR] Failures:
[ERROR]   AddCommandTest.addAsManangedDependencyForPomPackaging:162->BaseTest.assertXpath:78->BaseTest.assertXpath:82 Nodes matching ==> expected: <1> but was: <0>
[ERROR]   AddCommandTest.addToScope:75 expected: <0> but was: <2>
[ERROR]   AddCommandTest.addToScope:75 expected: <0> but was: <2>
[ERROR]   AddCommandTest.addToScope:75 expected: <0> but was: <2>
[ERROR]   AddCommandTest.addToScope:75 expected: <0> but was: <2>
[ERROR]   AddCommandTest.addToScope:75 expected: <0> but was: <2>
[ERROR]   AddCommandTest.addToScope:75 expected: <0> but was: <2>
[ERROR]   AddCommandTest.failIfDuplicateMultiple:117 expected: <1> but was: <2>
[ERROR]   AddCommandTest.failIfDuplicateSingle:94 expected: <1> but was: <2>
[ERROR]   AddCommandTest.versionAutoResolution:277 expected: <0> but was: <2>
[ERROR]   IdCommandTest.shouldAddParentIfAPomProjectIsFoundAboveTheDirectoryTree:241->BaseTest.assertXpath:78->BaseTest.assertXpath:82 Nodes matching ==> expected: <1> but was: <0>
[ERROR]   RemoveCommandTest.success:79 expected: <0> but was: <2>
[ERROR]   SearchCommandTest.sortByTimestampWhenGroupIdAndArtifactIdSpecified:82 expected: <Found 4
g:a:4                                                                                                          a day ago
g:a:3                                                                                                         3 days ago
g:a:2                                                                                                        3 years ago
g:a:1                                                                                                        6 years ago> but was: <Found 4
g:a:4                                                                                                          a day ago
g:a:3                                                                                                         3 days ago
g:a:2                                                                                                        3 years ago
g:a:1                                                                                                        6 years ago>
[ERROR]   SetParentCommandTest.test:51 Exit code ==> expected: <0> but was: <2>
[ERROR] Errors:
[ERROR]   AddCommandTest.shouldSuccesfullyAddOptional:463->BaseTest.assertXpath:78->BaseTest.assertXpath:82->BaseTest.evalXpath:73 » Runtime java.nio.file.NoSuchFileException: C:\Users\KRACHY~1\AppData\Local\Temp\junit3692129978000879468\pom.xml
[INFO]
[ERROR] Tests run: 71, Failures: 14, Errors: 1, Skipped: 0
```

So I hopped into WSL and did a `mvn verify` there with success. I don't know that I can trust this code when I need to run it on Windows. I have not taken the time to review the actual tests to see if I can spot why they failed.

--- a/src/main/java/com/github/andirady/pomcli/Main.java
+++ b/src/main/java/com/github/andirady/pomcli/Main.java
@@ -21,6 +21,7 @@ import java.io.UncheckedIOException;
 import java.net.URI;
 import java.nio.file.FileSystems;
 import java.nio.file.Files;
+import java.nio.file.InvalidPathException;
 import java.nio.file.Path;
 import java.util.Map;
 import java.util.Optional;
@@ -113,16 +114,20 @@ public class Main {
     }
 
     static Dependency stringToDependency(String s) {
-        var path = Path.of(s);
-        if (Files.isRegularFile(path)) {
-            var filename = path.getFileName().toString();
-            if (filename.endsWith(".jar")) {
-                return readCoordFromJarFile(path);
-            } else if (filename.endsWith(".xml")) {
-                return readCoordFromPomFile(path);
+        try {
+            var path = Path.of(s);
+            if (Files.isRegularFile(path)) {
+                var filename = path.getFileName().toString();
+                if (filename.endsWith(".jar")) {
+                    return readCoordFromJarFile(path);
+                } else if (filename.endsWith(".xml")) {
+                    return readCoordFromPomFile(path);
+                }
+            } else if (Files.isDirectory(path)) {
+                return readCoordFromPomFile(path.resolve("pom.xml"));
             }
-        } else if (Files.isDirectory(path)) {
-            return readCoordFromPomFile(path.resolve("pom.xml"));
+        } catch (InvalidPathException ignored) {
+
         }
 
         var d = new Dependency();

--- a/src/main/java/com/github/andirady/pomcli/converter/ParentConverter.java
+++ b/src/main/java/com/github/andirady/pomcli/converter/ParentConverter.java
@@ -18,6 +18,7 @@ package com.github.andirady.pomcli.converter;
 import java.io.IOException;
 import java.io.UncheckedIOException;
 import java.nio.file.Files;
+import java.nio.file.InvalidPathException;
 import java.nio.file.Path;
 
 import org.apache.maven.model.Parent;
@@ -33,43 +34,47 @@ public class ParentConverter implements ITypeConverter<Parent> {
 
     @Override
     public Parent convert(String spec) {
-        var path = Path.of(spec);
-        if (Files.exists(path)) {
-            path = Files.isDirectory(path) ? path.resolve("pom.xml") : path;
+        try {
+            var path = Path.of(spec);
+            if (Files.exists(path)) {
+                path = Files.isDirectory(path) ? path.resolve("pom.xml") : path;
 
-            var pomReader = new DefaultModelReader(null);
-            try (var is = Files.newInputStream(path)) {
-                var pom = pomReader.read(is, null);
-                if (!"pom".equals(pom.getPackaging())) {
-                    throw new IllegalArgumentException("The specified parent is not using pom packaging");
-                }
-
-                var g = pom.getGroupId();
-                var v = pom.getVersion();
-
-                var grandParent = pom.getParent();
-                if (grandParent != null) {
-                    if (g == null || g.isBlank()) {
-                        g = grandParent.getGroupId();
+                var pomReader = new DefaultModelReader(null);
+                try (var is = Files.newInputStream(path)) {
+                    var pom = pomReader.read(is, null);
+                    if (!"pom".equals(pom.getPackaging())) {
+                        throw new IllegalArgumentException("The specified parent is not using pom packaging");
                     }
-                    if (v == null || g.isBlank()) {
-                        v = grandParent.getVersion();
+
+                    var g = pom.getGroupId();
+                    var v = pom.getVersion();
+
+                    var grandParent = pom.getParent();
+                    if (grandParent != null) {
+                        if (g == null || g.isBlank()) {
+                            g = grandParent.getGroupId();
+                        }
+                        if (v == null || g.isBlank()) {
+                            v = grandParent.getVersion();
+                        }
                     }
+
+                    if (g == null || v == null || g.isBlank() || v.isBlank()) {
+                        throw new TypeConversionException(path + " is an invalid pom.");
+                    }
+
+                    var parent = new Parent();
+                    parent.setGroupId(g);
+                    parent.setArtifactId(pom.getArtifactId());
+                    parent.setVersion(v);
+
+                    return parent;
+                } catch (IOException e) {
+                    throw new UncheckedIOException(e);
                 }
-
-                if (g == null || v == null || g.isBlank() || v.isBlank()) {
-                    throw new TypeConversionException(path + " is an invalid pom.");
-                }
-
-                var parent = new Parent();
-                parent.setGroupId(g);
-                parent.setArtifactId(pom.getArtifactId());
-                parent.setVersion(v);
-
-                return parent;
-            } catch (IOException e) {
-                throw new UncheckedIOException(e);
             }
+        } catch (InvalidPathException ignored) {
+
         }
 
         var querySpec = QuerySpec.of(spec);

--- a/src/test/java/com/github/andirady/pomcli/IdCommandTest.java
+++ b/src/test/java/com/github/andirady/pomcli/IdCommandTest.java
@@ -23,6 +23,7 @@ import static org.junit.jupiter.api.Assertions.assertTrue;
 import java.io.IOException;
 import java.io.PrintWriter;
 import java.io.StringWriter;
+import java.nio.file.FileSystems;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.List;
@@ -237,7 +238,7 @@ class IdCommandTest extends BaseTest {
         underTest.execute("id", "-f", pomPath.toString(), ".");
         LOG.fine(Files.readString(pomPath));
 
-        var expr = "/project/parent[groupId='a' and artifactId='a' and version='1' and relativePath='../..']";
+        var expr = "/project/parent[groupId='a' and artifactId='a' and version='1' and relativePath='.." + FileSystems.getDefault().getSeparator() + "..']";
         assertXpath(pomPath, expr, 1);
         assertEquals("jar a:c:1", out.toString().trim());
     }

--- a/src/test/java/com/github/andirady/pomcli/SearchCommandTest.java
+++ b/src/test/java/com/github/andirady/pomcli/SearchCommandTest.java
@@ -15,7 +15,7 @@
  */
 package com.github.andirady.pomcli;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertLinesMatch;
 
 import java.io.PrintWriter;
 import java.io.StringWriter;
@@ -79,6 +79,6 @@ class SearchCommandTest {
                 g:a:1                                                                                                    @|fg(238)     6 years ago|@""");
         var actual = out.toString();
 
-        assertEquals(expected, actual);
-	}
+        assertLinesMatch(expected.lines(), actual.lines());
+    }
 }

